File: src/main/java/com/pubnub/api/models/server/files/FileUploadRequestDetails.java
package com.pubnub.api.models.server.files;

import com.pubnub.api.models.consumer.files.PNFile;
import lombok.Data;

import java.util.List;

@Data
public class FileUploadRequestDetails {
    private final Integer status;
    private final PNFile data;
    private final String url;
    private final String method;
    private final String expirationDate;
    private final FormField keyFormField;
    private final List<FormField> formFields;
}

File: src/main/kotlin/com/pubnub/api/models/server/files/FileUploadRequestDetails.kt
package com.pubnub.api.models.server.files

import com.pubnub.api.models.consumer.files.PNFile

data class FileUploadRequestDetails(
    val status: Int,
    val data: PNFile,
    val url: String,
    val method: String,
    val expirationDate: String,
    val keyFormField: FormField,
    val formFields: List<FormField>
)
